src/HubTextNotesLayer.js

```diff
@@ -66,7 +67,16 @@ const HubTextNotesLayer = Layer.createSubclass({
 
   // retrieve note by referenced graphic
   findNoteForGraphic (graphic) {
-    return this.hubNotes.find(note => note.graphic === graphic);
+    // first look for a note with same graphic identity, should work for GraphicsLayer
+    let note = this.hubNotes.find(note => note.graphic === graphic);
+    if (note) {
+      return note;
+    }
+
+    // then fallback to comparing by object id, necessary for features from a FeatureLayer.hitTest()
+    return this.hubNotes
+      .filter(note => note.graphic.getObjectId() != null)
+      .find(note => note.graphic.getObjectId() === graphic.getObjectId());
```

Comment:
Would this suffice when dealing with features from multiple feature layers? Would it make sense to store `layerId` as well?

Reply:
Thank you, this is a good call... need to do some explicit tests w/multiple layers but I think you are right we will want an additional check here.